app/models/live_metric.rb

```diff
@@ -3,6 +3,10 @@ class LiveMetric < ActsAsArModel
 
   class LiveMetricError < RuntimeError; end
 
+  def self.virtual_attribute?(c)
+    super || c =~ /^mw_/
```

Comment:
I think this one won't work well, as it is assuming that all metrics should start with the "mw_" prefix (which might not be true).
Basically  LiveMetric object has a fixed field

```
set_columns_hash(:timestamp => :datetime) 
```
And rest of the fields are prepared on:

```
  def self.filter_and_fetch_metrics(resource, filter, start_time, end_time, interval_name)
    filtered = resource.metrics_available.select { |metric| filter.nil? || filter.include?(metric[:name]) }
    filtered.each { |metric| set_columns_hash(metric[:name] => :float) }
    fetch_live_metrics(resource, filtered, start_time, end_time, interval_name)
  end
```
Which is basically doing is to add the fields that are previously parsed from a config file.

One question that perhaps might simplify, in case of LiveMetric objects, all fields are virtual, so why don't return a true ?

A complete end-to-end flow test with reports is not available, but there are some tests under

```
spec/models/manageiq/providers/hawkular/middleware_manager/middleware_*_spec.rb

```

These ones invoke LiveMetric but not from the generator.rb, sorry.

If there is an example of integration test I can take a look and put it on my todo list.